build script: missing openssl deps + pipefail silent failure on make | tee

## Bug 1: `openssl` and `openssl-devel` missing from `dnf install`

**Function:** `install_deps`

The `dnf install` block does not include `openssl` or `openssl-devel`. The kernel build needs OpenSSL to sign modules and generate `certs/signing_key.pem`. Without these packages the build fails silently with:

```
make[2]: *** [certs/Makefile:...] Error 127
```

`Error 127` means the `openssl` binary was not found on `PATH`. Because `set -e` is defeated by the `tee` pipe (see Bug 2 below), the build continues and the failure is not surfaced until much later—or is missed entirely.

**Fix:** add `openssl openssl-devel` to the `sudo dnf install -y` block.

---

## Bug 2: `set -e` without `pipefail` silently ignores `make` failures

**Functions:** `build_kernel`, `install_kernel`

Both functions run `make` as the left-hand side of a pipeline:

```bash
make -j$(nproc) 2>&1 | tee -a "$LOG_FILE"
sudo make modules_install 2>&1 | tee -a "$LOG_FILE"
# … etc.
```

With `set -e` but **without** `set -o pipefail`, bash evaluates the exit status of the last command in the pipeline—`tee`—which almost always exits 0. A failed `make` therefore does **not** trigger `set -e`, and the script continues as if the build succeeded.

The script already defines a correct `log_and_run` helper that captures `${PIPESTATUS[0]}` and returns it to the caller, but `build_kernel` and `install_kernel` bypass it and pipe `make` directly through `tee`.

**Fix:**
1. Change `set -e` → `set -eo pipefail` at the top of the script.
2. Replace every `make ... 2>&1 | tee -a "$LOG_FILE"` call in `build_kernel` and `install_kernel` with `log_and_run make ...` (using the existing helper).

---

## Environment

Tested on: MacBookPro17,1 · Apple M1 · Fedora Asahi Remix 43

--- a/asahi-fairydust-build.sh
+++ b/asahi-fairydust-build.sh
@@ -36,7 +36,7 @@
 #   - This script is provided as-is with no warranty
 # =============================================================================
 
-set -e
+set -eo pipefail
 
 # --- Configuration ---
 CLONE_DIR="$HOME/linux-fairydust"
@@ -145,6 +145,7 @@ install_deps() {
         gcc gcc-c++ make bc bison flex elfutils-libelf-devel \
         ncurses-devel python3 zlib-devel libuuid-devel dwarves \
         xz zstd clang llvm lld git \
+        openssl openssl-devel \
         rust rust-std-static bindgen-cli \
         2>&1 | tee -a "$LOG_FILE"
 
@@ -306,7 +307,7 @@ build_kernel() {
 
     cd "$CLONE_DIR"
 
-    make -j$(nproc) 2>&1 | tee -a "$LOG_FILE"
+    log_and_run make -j$(nproc)
 
     ok "Kernel build completed at: $(date)"
 }
@@ -323,20 +324,20 @@ install_kernel() {
 
     # Install modules, DTBs, VDSO
     info "Installing modules..."
-    sudo make INSTALL_MOD_STRIP=1 modules_install 2>&1 | tee -a "$LOG_FILE"
+    log_and_run sudo make INSTALL_MOD_STRIP=1 modules_install
 
     info "Installing DTBs..."
-    sudo make dtbs_install 2>&1 | tee -a "$LOG_FILE"
+    log_and_run sudo make dtbs_install
 
     info "Installing VDSO..."
-    sudo make vdso_install 2>&1 | tee -a "$LOG_FILE"
+    log_and_run sudo make vdso_install
 
     # Create DTB symlink that Fedora expects
     sudo ln -sf "/boot/dtbs/$KVER" "/usr/lib/modules/$KVER/dtb"
 
     # Install kernel image
     info "Installing kernel image..."
-    sudo make install 2>&1 | tee -a "$LOG_FILE"
+    log_and_run sudo make install
 
     ok "Kernel installed: $KVER"
 }

